lib/bundler/rubygems_integration.rb

```diff
@@ -39,7 +39,12 @@ def loaded_specs(name)
     end
 
     def mark_loaded(spec)
-      Gem.loaded_specs[spec.name] = spec
+      name = spec.name
+      if existing = Gem.loaded_specs[name]
+        existing.activated = false
```

Comment:
probably needs to make sure that `existing.respond_to? :activated` if we want to merge this :/